### PR TITLE
Jv/rox 9961 create nightly collector circleci job

### PR DESCRIPTION
**Pull request by JoukoVirtanen**
## Description

Create a nightly job

## Checklist
- [x] Investigated and inspected CI test results
- [x] Updated documentation accordingly

### CODE REVIEW

File: .circleci/config.yml
Comment (JoukoVirtanen) [inline .circleci/config.yml:1722]: - [ ] Change to master before merging

File: .circleci/config.yml
Comment (Molter73) [inline .circleci/config.yml:1007]: We probably also want to run all integration tests for all tags, just to double check when we cherry-pick commits onto an existing tag.

File: .circleci/config.yml
Comment (Molter73) [inline .circleci/config.yml:1448]: The name for this anchor is not very descriptive, can we try to rename it to something like `fullWorkflow` or `completeWorkflow`?

File: .circleci/config.yml
Comment (Molter73) [reply .circleci/config.yml:1722]: Probably not something to address in this PR, but since we had issues a few months back with drivers not being built for legacy versions of collector it might be worth it to add a few tags alongside master here for future proving.

**Review by Molter73 — COMMENTED** [review]
The integration test gather data step is failing, I think it might be related to some recent changes on how that information is gathered, can we try rebasing the PR to see if it works properly?

Other than this and a few additional comments, the PR is looking very good!

File: .circleci/config.yml
Comment (JoukoVirtanen) [reply .circleci/config.yml]: Done

File: .circleci/config.yml
Comment (JoukoVirtanen) [reply .circleci/config.yml]: Done

File: .circleci/config.yml
Comment (JoukoVirtanen) [inline .circleci/config.yml:63]: Uncomment before merging

File: .circleci/config.yml
Comment (JoukoVirtanen) [inline .circleci/config.yml:1346]: Uncomment before merging

File: .circleci/test-scripts/integration-test/test-check-whether-to-run-job.sh
Comment (Molter73) [inline .circleci/test-scripts/integration-test/test-check-whether-to-run-job.sh:1]: This is a nitpick and maybe something to address in a follow up PR, but I would much rather have the `TestInput.txt` file for this be in something like yaml or json so we don't have to go around trying to guess what each word in it is supposed to mean, as well as probably having this test be written in python to make it easier to process one of those formats.

File: .circleci/config.yml
Comment (Molter73) [inline .circleci/config.yml:917]: As far as I can tell `dockerized` is not being used.
```suggestion
```

File: .circleci/integration-test/10-check-whether-to-run-job.sh
Comment (Molter73) [inline .circleci/integration-test/10-check-whether-to-run-job.sh:14]: Why are we using `echo` to output the result of the script and not using the exit code directly?

File: .circleci/config.yml
Comment (Molter73) [inline .circleci/config.yml:1022]: This script is only called here with its only parameter set to `true`, could we remove this and cleanup both `.circleci/test-scripts/baseline/compare-and-update-baseline.sh` and `.circleci/test-scripts/baseline/main.py`, since it looks like this is now its default behavior and I don't see a point complicating the code if it is not going to be used.

**Review by Molter73 — COMMENTED** [review]
Just have a few minor comments, I think I might be missing some context for `.circleci/test-scripts/baseline/main.py` so I would pretty much prefer it if @erthalion took a look before merging the PR.

Also, seems the PR is a little outdated and is not running the `ubi` steps, could you rebase it since those have been fixed?

File: .circleci/config.yml
Comment (JoukoVirtanen) [reply .circleci/config.yml:917]: Done

File: .circleci/config.yml
Comment (JoukoVirtanen) [reply .circleci/config.yml:1022]: Done

File: .circleci/test-scripts/baseline/main.py
Comment (Molter73) [inline .circleci/test-scripts/baseline/main.py:268]: This function seems to be unused.

File: .circleci/test-scripts/integration-test/test-check-whether-to-run-job.sh
Comment (Molter73) [inline .circleci/test-scripts/integration-test/test-check-whether-to-run-job.sh:18]: Sorry I missed this in my previous reviews, but maybe we can use `read -a` to read values directly into an array and make it easier to assign these variables?
```suggestion
while read -r -a line; do
    dockerized="${line[1]}"
    image_family="${line[2]}"
    trigger_source="${line[3]}"
    BRANCH="${line[4]}"
    TAG="${line[5]}"
    expected_should_run="${line[6]}"
```

**Review by Molter73 — COMMENTED** [review]
I think the PR is looking good. I do want to mention that I encountered the same problem you did with the new performance script when trying to update a VM in out integration tests.

Maybe we can remove your changes from `.circleci/test-scripts/baseline/main.py` and stack this PR on top of #673? WDYT?

File: .circleci/test-scripts/integration-test/test-check-whether-to-run-job.sh
Comment (JoukoVirtanen) [reply .circleci/test-scripts/integration-test/test-check-whether-to-run-job.sh]: Done

File: .circleci/test-scripts/baseline/main.py
Comment (JoukoVirtanen) [reply .circleci/test-scripts/baseline/main.py]: Removed

**Comment by JoukoVirtanen** [comment]
> I think the PR is looking good. I do want to mention that I encountered the same problem you did with the new performance script when trying to update a VM in out integration tests.
> 
> Maybe we can remove your changes from `.circleci/test-scripts/baseline/main.py` and stack this PR on top of #673? WDYT?

I have no objection to that., but I am not 100% sure how to do that.

**Comment by JoukoVirtanen** [comment]
> > I think the PR is looking good. I do want to mention that I encountered the same problem you did with the new performance script when trying to update a VM in out integration tests.
> > Maybe we can remove your changes from `.circleci/test-scripts/baseline/main.py` and stack this PR on top of #673? WDYT?
> 
> I have no objection to that., but I am not 100% sure how to do that.

I reverted my changes to main.py. Do you just want me wait until you have merged your PR into master and then I can rebase to get your changes to main.py?

**Comment by Molter73** [comment]
There are a few ways you can go about it, you could wait for my PR to be merged, but if you want to test your changes with mine before that you can:
- Change the base branch in this PR from `master` to `mauro/add-ubuntu-2204-tests`
- Rebase your changes on top of `mauro/add-ubuntu-2204-tests` (you might be able to do so with `smart-rebase` but I wouldn't know).
- Check locally that all your changes are still showing up correctly and force push the branch.

Keep in mind that doing this will mean that, once I merge my PR, you will probably need to rebase your changes on top of master once more, (because we squash merge and the commit history is lost, which causes git/GH to be a little confused).

**Review by Molter73 — APPROVED** [review]
LGTM!

**Pull request merged**
